src/nvim/drawscreen.c

```diff
@@ -629,7 +629,12 @@ int update_screen(void)
         did_one = true;
         start_search_hl();
       }
+
+      int win_update_type = wp->w_redr_type;
       win_update(wp, &providers);
+      if (win_update_type) {
```

Comment:
Is this check needed? This is only reached when `wp->w_redr_type != 0`

Reply:
Good catch! No it appears like it isn’t. I must have missed the outer block that already does the check, I’m not used two reading code with just two space indentation.

Reply:
This should be fixed now